evmrpc/filter.go

```diff
@@ -0,0 +1,199 @@
+package evmrpc
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/eth/filters"
+	abci "github.com/tendermint/tendermint/abci/types"
+	rpcclient "github.com/tendermint/tendermint/rpc/client"
+	"github.com/tendermint/tendermint/rpc/coretypes"
+)
+
+type filter struct {
+	fc      filters.FilterCriteria
+	cursors map[common.Address]string
+}
+
+type FilterAPI struct {
+	tmClient     rpcclient.Client
+	nextFilterID uint64
+	filters      map[uint64]filter
+}
+
+func NewFilterAPI(tmClient rpcclient.Client) *FilterAPI {
+	filters := make(map[uint64]filter)
+	return &FilterAPI{tmClient: tmClient, nextFilterID: 1, filters: filters}
+}
+
+func (a *FilterAPI) NewFilter(
+	ctx context.Context,
+	crit filters.FilterCriteria,
+) (*uint64, error) {
+	curFilterID := a.nextFilterID
+	a.nextFilterID++
```

Comment:
we need to protect this section behind a mutex since concurrent calls are possible

Reply:
updated, along with timeouts change